bgs/bgs.go

```diff
@@ -87,6 +88,10 @@ type BGS struct {
 
 	// Management of Compaction
 	compactor *Compactor
+
+	apiKeys        atomic.Pointer[ApiKeys]
```

Comment:
Is there a reason we use this instead of a lock?

Reply:
We don't use these atomic semantics anywhere else and I'm a bit wary of throwing them in here now over using a standard mutex pattern.

Reply:
Ideally atomics are faster; there are a few large system NUMA cases where they are less good than usual, but I'm not sure they're ever slower than sync.Mutex ? I think in the optimistic case where there is no contention they're much faster.
They don't provide fairness of unlock to multiple lockers like sync.Mutex, but that doesn't matter here.